src/ol_openedx_canvas_integration/client.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import pytz
+from urllib.parse import urlencode, urljoin, urlparse, parse_qs
+import requests
+
+from django.conf import settings
+
+from ol_openedx_canvas_integration.constants import DEFAULT_ASSIGNMENT_POINTS
+
+
+log = logging.getLogger(__name__)
+
+
+class CanvasClient:
+    def __init__(self, canvas_course_id):
+        self.session = self.get_canvas_session()
+        self.canvas_course_id = canvas_course_id
+
+    @staticmethod
+    def get_canvas_session():
+        """
+        Create a request session with the access token
+        """
+        session = requests.Session()
+        session.headers.update({
+            "Authorization": "Bearer {token}".format(token=settings.CANVAS_ACCESS_TOKEN)
```

Comment:
I think we can persist canvas credentials in encrypted field of django model instead of using them from settings

Reply:
@arslanashraf7 can you check with @mitodl/devops for their preference for managing the secrets?

Reply:
Sure, Currently we are injecting them from the vault https://github.com/search?q=org%3Amitodl+CANVAS_ACCESS_TOKEN&type=code. I'll talk to the ops guys if they want to go with this approach or the existing vault approach is better.

Reply:
It's fine to keep it as a settings attribute because we can take the same approach of managing the value through the configuration template.